### ISSUE TITLE
The shim exe should not have an icon if the original program doesn't have one

### ISSUE THREAD

**Issue by Jack-Works** [chocolatey/home#355]
### What You Are Seeing?

![image](https://user-images.githubusercontent.com/5390719/86236179-3bd8ae00-bbcc-11ea-87f1-957e6780b2aa.png)

### What is Expected?

The original program doesn't have an icon so it's using the Windows default icon. The shim should follow the original program.

### How Did You Get This To Happen? (Steps to Reproduce)

> choco install deno

**Comment by pauby** [comment]
As was mentioned in the [comment above](https://github.com/chocolatey/home/issues/355#issuecomment-3007749713), we show the "shim" icon to indicate it has been shimmed.  

You can indicate the icon you want to use with the `--icon-path` option:

```pwsh
$ C:\ProgramData\chocolatey\tools\shimgen.exe

ShimGen generates an executable that will call a file relative to its location. Think of it as batch redirect 2.0.
To use shimgen, supply an output path for the shim executable, relative path to executable, and whether it is a GUI you are calling
Example: shimgen --output=bob --path="the\ relative \path"
Example: shimgen -o="..\bob.exe" -p="the\ relative \path" --gui
  -?, --help, -h             Prints out the options.
  -o, --output=VALUE         REQUIRED: output - The is the name of the shim
                               (e.g. rh.exe for rh.exe). This can also contain
                               the path to the output file
  -p, --path=VALUE           REQUIRED: path - The path to the executable. Can
                               be relative or absolute.
  -i, --iconpath=VALUE       : path of executable for icon - The path to the
                               executable for resolving the icon. Can be
                               relative or absolute (suggest absolute).
  -c, --command=VALUE        command - The command you want to pass to the
                               executable.
      --gui                  GUI - This instructs ShimGen that the underlying
                               executable is a GUI and it should not wait for
                               it to finish execution.
      --debug                debug - This instructs ShimGen to write out all
                               messages. Defaults to false.
```

**Comment by ThomasHeinrichSchmidt** [comment]
That's nice. So next time my shimmed exe icons get reset to the shim icon (when package was updated), I'l try to use  
`C:\ProgramData\chocolatey\tools\shimgen.exe --path="c:\ProgramData\chocolatey\lib\exiftool\tools\exiftool-13.31_64\exiftool_files\perl.exe" --output="c:\ProgramData\chocolatey\bin\perl.exe" --iconpath="c:\ProgramData\chocolatey\lib\exiftool\tools\exiftool-13.31_64\exiftool_files\perl.exe"`

When I try this now there is an error message `
[WARN ] Could not extract icon from associated program. Using default. Error:
  Selected Icon is invalid
`
And this is probably the reason for all my "default" shim icons. As I said I was able to manually extract icons from all executables I'm using. So `shimgen.exe` with the` --icon-path` option is not the solution for me.

**Comment by pauby** [comment]
What I meant to indicate is that you can use the `--icon-path` option to use the icon in the executable you point to. Not being able to extract the icon is a separate issue that we are discussing over [here](https://github.com/chocolatey/home/issues/354#issuecomment-3007946764).

As we intentionally use the default shim icon for executables that do not have an icon that we can extract, I'm going to fo ahead and close this. We can reopen it again if necessary.

**Comment by ThomasHeinrichSchmidt** [comment]
Yes I understand, all good, looking forward to `shimgen.exe` being able to process all executables.
